model/DataObject.php

```diff
@@ -1744,44 +1745,79 @@ public function getManyManyComponents($componentName, $filter = "", $sort = "",
+	}
+
+	/**
 	 * Return the class of a one-to-one component.  If $component is null, return all of the one-to-one components and
 	 * their classes. If the selected has_one is a polymorphic field then 'DataObject' will be returned for the type.
 	 *
 	 * @param string $component Name of component
-	 *
 	 * @return string|array The class of the one-to-one component, or an array of all one-to-one components and their
 	 *                      classes.
 	 */
-	public function has_one($component = null) {
-		$classes = ClassInfo::ancestry($this);
+	public function hasOne($component = null) {
+		if($component) {
+			return $this->hasOneComponent($component);
+		}
 
+		$classes = ClassInfo::ancestry($this);
 		foreach($classes as $class) {
 			// Wait until after we reach DataObject
 			if(in_array($class, array('Object', 'ViewableData', 'DataObject'))) continue;
 
-			if($component) {
-				$hasOne = Config::inst()->get($class, 'has_one', Config::UNINHERITED);
-
-				if(isset($hasOne[$component])) {
-					return $hasOne[$component];
+			$newItems = (array)Config::inst()->get($class, 'has_one', Config::UNINHERITED);
+			// Validate the data
+			foreach($newItems as $k => $v) {
+				if(!is_string($k) || is_numeric($k) || !is_string($v)) {
+					user_error("$class::\$has_one has a bad entry: "
+					. var_export($k,true). " => " . var_export($v,true) . ".  Each map key should be a"
+					. " relationship name, and the map value should be the data class to join to.", E_USER_ERROR);
 				}
-			} else {
-				$newItems = (array)Config::inst()->get($class, 'has_one', Config::UNINHERITED);
-				// Validate the data
-				foreach($newItems as $k => $v) {
-					if(!is_string($k) || is_numeric($k) || !is_string($v)) {
-						user_error("$class::\$has_one has a bad entry: "
-						. var_export($k,true). " => " . var_export($v,true) . ".  Each map key should be a"
-						. " relationship name, and the map value should be the data class to join to.", E_USER_ERROR);
-					}
-				}
-				$items = isset($items) ? array_merge($newItems, (array)$items) : $newItems;
 			}
+			$items = isset($items) ? array_merge($newItems, (array)$items) : $newItems;
 		}
+
 		return isset($items) ? $items : null;
 	}
 
 	/**
+	 * Return data for a specific has_one component.
+	 * @param string $component
+	 * @return array|null
+	 */
+	public function hasOneComponent($component) {
+		$classes = ClassInfo::ancestry($this);
+		foreach($classes as $class) {
+			// Wait until after we reach DataObject
+			if(in_array($class, array('Object', 'ViewableData', 'DataObject'))) continue;
+
+			$hasOne = Config::inst()->get($class, 'has_one', Config::UNINHERITED);
+
+			if(isset($hasOne[$component])) {
+				return $hasOne[$component];
+			}
+		}
+	}
+
+	/**
+	 * @deprecated 4.0 Method has been renamed to belongsTo()
+	 * @param string $component
+	 * @param bool $classOnly
+	 * @return array|null
+	 */
+	public function belongs_to($component = null, $classOnly = true) {
```

Comment:
Hm, is `$classOnly = false` used by functions to resolve ambiguous relationships where dot notation is used to find the reverse relation?